galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager/src/main/java/dev/galasa/cicsts/ITsqFactory.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.cicsts;
+
+import javax.validation.constraints.NotNull;
+public interface ITsqFactory {
+
+      /**
+       * Create a new ITsq object with recoverable status   
+       * @param queueName TSQ name
+       * @param recoverable true for recoverable and false for non-recoverable
+       * @return ITsq object
+       * @throws TsqException if there is a problem in creating the ITsq object
+       */
+      public ITsq createQueue(@NotNull String queueName, @NotNull boolean recoverable) throws TsqException;
```

Comment:
boolean parameters can never be null, so @NotNull for a boolean flag is unnecessary. Also boolean parameters should really have the 'is' prefix in the name. isRecoverable ...etc.

Reply:
Updated the code.